client/services.go

```diff
@@ -58,3 +60,32 @@ func (client *Client) doMultiServiceAction(actionName string, services []string)
 	}
 	return client.doAsyncFull("POST", "/v1/services", nil, headers, bytes.NewBuffer(data))
 }
+
+// ServicesOptions are the filtering options for querying services.
+type ServicesOptions struct {
+	// Names is the list of service names to query for. If slice is nil or
+	// empty, fetch information for all services.
+	Names []string
+}
+
+// ServiceInfo holds status information for a single service.
+type ServiceInfo struct {
+	Name    string `json:"name"`    // service name
+	Default string `json:"default"` // default state: "start" or "stop"
+	Status  string `json:"status"`  // "active", "inactive", or "error"
+	Message string `json:"message"` // message (e.g., error) or empty string
```

Comment:
What kind of messages would we have here?  In particular, when will we display something here that is not an error, and when it is an error how do we tell what the error is without listing the entire log?  The concern is two fold: the listing command should remain a listing command. Even a short message, with 50 chars, will turn the columns into a mess, while often not giving enough context for the error. So we need to find out how we both make it easily introspectable, while providing a good experience for people navigating through it.

As usual in those cases, we should probably do less before we know what we want. Happy to discuss this further elsewhere for more bandwidth.

Reply:
Yeah, I think you're right. I've removed for now as I wasn't currently populating this field anyway. We can always add it later when we've defined it / got more experience with Pebble.